ydb/core/kqp/common/kqp_tx_manager.cpp

```diff
@@ -235,11 +271,14 @@ class TKqpTransactionManager : public IKqpTransactionManager {
         for (auto& [shardId, shardInfo] : ShardsInfo) {
             if ((shardInfo.Flags & EAction::WRITE)) {
                 ReceivingShards.insert(shardId);
-                if (IsVolatile()) {
-                    SendingShards.insert(shardId);
-                }
-                if (shardInfo.IsOlap) {
-                    sendingColumnShardsSet.insert(shardId);
+                
+                if (!shardInfo.IsTopic) {
```

Comment:
Don't need it. 
IsVolatile = false in presence of topics.
Olap logic with columnShardArbiter should work fine with topics